src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,38 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  todo: Todo | null;
+  setVisible: React.Dispatch<React.SetStateAction<boolean>>;
+  setTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+};
+
+export const TodoModal: React.FC<Props> = ({ setVisible, todo, setTodo }) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setIsLoading(false);
+    }, 1000);
+
+    if (todo) {
+      getUser(todo.userId).then(setUser);
+    }
+  }, []);
```

Comment:
Consider adding `todo` to the dependency array of the `useEffect` hook to ensure it re-runs when the `todo` prop changes. This will allow the component to fetch the correct user data for the new `todo`.